src/ImageSearchModal.jsx

```diff
@@ -15,238 +15,203 @@ import cockpit from 'cockpit';
+        if (activeConnection)
+            activeConnection.close();
+        setSearchFinished(false);
 
         // Do not call the SearchImage API if the input string  is not at least 2 chars,
         // unless Enter is pressed, which should force start the search.
         // The comparison was done considering the fact that we miss always one letter due to delayed setState
-        if (this.state.imageIdentifier.length < 2 && !forceSearch)
+        if (imageIdentifier.length < 2 && !forceSearch)
             return;
 
-        this.setState({ searchInProgress: true });
+        setSearchInProgress(true);
 
-        this.activeConnection = rest.connect(client.getAddress(this.state.isSystem), this.state.isSystem);
-        let registries = Object.keys(this.props.registries).length !== 0 ? [this.state.registry] : fallbackRegistries;
+        let queryRegistries = searchRegistries;
+        if (searchRegistry !== "") {
+            queryRegistries = [searchRegistry];
+        }
         // if a user searches for `docker.io/cockpit` let podman search in the user specified registry.
-        if (this.state.imageIdentifier.includes('/')) {
-            registries = [""];
+        if (imageIdentifier.includes('/')) {
+            queryRegistries = [""];
         }
 
-        const searches = registries.map(rr => {
+        const searches = queryRegistries.map(rr => {
             const registry = rr.length < 1 || rr[rr.length - 1] === "/" ? rr : rr + "/";
-            return this.activeConnection.call({
+            return activeConnection.call({
                 method: "GET",
                 path: client.VERSION + "libpod/images/search",
                 body: "",
                 params: {
-                    term: registry + this.state.imageIdentifier
+                    term: registry + imageIdentifier
                 }
             });
         });
 
         Promise.allSettled(searches)
                 .then(reply => {
-                    if (reply && this._isMounted) {
+                    if (reply) {
                         let results = [];
-                        let dialogError = "";
-                        let dialogErrorDetail = "";
 
                         for (const result of reply) {
                             if (result.status === "fulfilled") {
                                 results = results.concat(JSON.parse(result.value));
                             } else {
-                                dialogError = _("Failed to search for new images");
-                                dialogErrorDetail = result.reason ? cockpit.format(_("Failed to search for images: $0"), result.reason.message) : _("Failed to search for images.");
+                                setDialogError(_("Failed to search for new images"));
+                                setDialogErrorDetail(result.reason ? cockpit.format(_("Failed to search for images: $0"), result.reason.message) : _("Failed to search for images."));
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230705-092041-36f39867-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)